internal/telemetry/client.go

```diff
@@ -340,35 +350,58 @@ func (c *client) Count(namespace Namespace, name string, value float64, tags []s
 // sent to the backend. Requests are sent in the background. Must be called
 // with c.mu locked
 func (c *client) flush() {
-	submissions := make([]*Request, 0, len(c.requests)+1)
+	submissions := make([]*Request, 0, len(c.requests)+2)
+
+	// copy over requests so we can do the actual submission without holding
+	// the lock. Zero out the old stuff so we don't leak references
+	for i, r := range c.requests {
+		submissions = append(submissions, r)
+		c.requests[i] = nil
+	}
+	c.requests = c.requests[:0]
+
 	if c.newMetrics {
 		c.newMetrics = false
-		r := c.newRequest(RequestTypeGenerateMetrics)
 		for namespace := range c.metrics {
-			payload := &Metrics{
+			// metrics can either be request type generate-metrics or distributions
+			gPayload := &Metrics{
+				Namespace: namespace,
+			}
+			dPayload := &Metrics{
 				Namespace: namespace,
 			}
 			for _, m := range c.metrics[namespace] {
-				s := Series{
-					Metric: m.name,
-					Type:   string(m.kind),
-					Tags:   m.tags,
-					Common: m.common,
+				if m.kind == MetricKindDist {
+					s := Series{
+						Metric: m.name,
+						Tags:   m.tags,
+						Common: m.common,
+					}
+					s.Points = [][2]float64{{m.ts, m.value}}
```

Comment:
Why not do this inline as part of the declaration on 375? (I might be missing something obvious :P )

Reply:
Inline works, I think I just missed this while i was refactoring the code thanks!